internal/ethapi/api.go

```diff
@@ -913,14 +924,49 @@ func (s *PublicBlockChainAPI) GetCandidates(ctx context.Context, epoch rpc.Epoch
 	} else {
 		log.Error("Undefined XDPoS consensus engine")
 	}
-	// Set masternode status
+
+	// Set all candidate to statusProposed
+	candidatesStatusMap := make(map[string]map[string]interface{}, len(candidates))
+	for _, candidate := range candidates {
+		candidatesStatusMap[candidate.Address.String()] = map[string]interface{}{
+			fieldStatus:   statusProposed,
+			fieldCapacity: candidate.Stake,
+		}
+	}
+
+	// Set masternodes to statusMasternode
 	for _, masternode := range masternodes {
-		if candidatesStatusMap[masternode.String()] != nil {
-			candidatesStatusMap[masternode.String()][fieldStatus] = statusMasternode
+		key := masternode.String()
+		if candidatesStatusMap[key] != nil {
+			candidatesStatusMap[key][fieldStatus] = statusMasternode
+		} else {
+			candidatesStatusMap[key] = map[string]interface{}{
+				fieldStatus:   statusMasternode,
+				fieldCapacity: -1,
+			}
+			log.Warn("Masternode is not candidate", "masternode", key, "checkpointNumber", checkpointNumber, "epoch", epoch, "epochNumber", epochNumber)
 		}
 	}
 
-	// Third, Get penalties list
+	var maxMasternodes int
+	if s.b.ChainConfig().IsTIPIncreaseMasternodes(block.Number()) {
+		maxMasternodes = common.MaxMasternodesV2
+	} else {
+		maxMasternodes = common.MaxMasternodes
+	}
+
+	if len(masternodes) >= maxMasternodes {
+		result[fieldCandidates] = candidatesStatusMap
+		return result, nil
+	}
+
+	if len(candidates) > maxMasternodes {
+		sort.Slice(candidates, func(i, j int) bool {
+			return candidates[i].Stake.Cmp(candidates[j].Stake) > 0
```

Comment:
In your description you say `>=0`, in code you use `>0`. Please make it the same.

Reply:
Yes, it should be `> 0`. I corrected it now.